src/popper/runner_host.py

```diff
@@ -281,24 +229,160 @@ def _find_container(self, cid):
 
         return None
 
-    def _update_with_engine_config(self, container_args):
-        """Given container arguments, it extends it so it includes options
-        obtained from the popper.config.Config.engine_opts property.
-        """
-        update_with = self._config.engine_opts
-        if not update_with:
+
+class PodmanRunner(StepRunner):
+
+    """Runs steps in podman on the local machine."""
+
+    def __init__(self, init_podman_client=True, **kw):
+        super(PodmanRunner, self).__init__(**kw)
+
+        self._spawned_containers = set()
+
+        if not init_podman_client:
+            return
+
+        try:
+            _, _, self._p_info = HostRunner._exec_cmd(["podman", "info"], logging=False)
+            self._p_version = HostRunner._exec_cmd(["podman", "version"], logging=False)
+        except Exception as e:
+            log.debug(f"Podman error: {e}")
+            log.fail("Unable to connect to podman, is it installed?")
+
+        log.debug(f"Podman info: {pu.prettystr(self._p_info)}")
+
+    def run(self, step):
+        """Executes the given step in podman."""
+        cid = pu.sanitized_name(step.id, self._config.wid)
+        container = self._find_container(cid)
+
+        if not container and self._config.reuse:
+            log.fail(
+                f"Cannot find an existing container for step '{step.id}' to be reused"
+            )
+
+        if container and not self._config.reuse and not self._config.dry_run:
+            cmd = ["podman", "rm", "-f", container]
+            HostRunner._exec_cmd(cmd, logging=False)
+            container = None
+
+        if not container and not self._config.reuse:
+            container = self._create_container(cid, step)
+
+        log.info(f"[{step.id}] podman start")
+
+        if self._config.dry_run:
+            return 0
+
+        self._spawned_containers.add(container)
+
+        cmd = ["podman", "start", "-a", container]
+        _, e, output, = HostRunner._exec_cmd(cmd)
+
+        cmd1 = ["podman", "inspect", "-f", "{{.State.ExitCode}}", container]
```

Comment:
this shouldn't be needed. Just doing `podman start` then checking for exit code should be enough

Reply:
Oh yes so when I do a `podman start`, the status code returned by HostRunner is always 0. It never throws an error in this case, which is why I have to inspect the container for the exit code.